### PR TITLE
Add an option to dwds to suppress emitting debug events

### PR DESCRIPTION
**Pull request by annagrin**
Flutter sends a stream of debug events to dwds for profiling information
that can slow down debugging for apps that refresh frames frequently.

Add an option to disable debug events as a workaround until we can improve
performance of communication between the injected client and dwds.

Workaround for: https://github.com/dart-lang/webdev/issues/1466

### CODE REVIEW

File: dwds/lib/src/handlers/injector.dart [hunk 1]
```diff
@@ -198,6 +205,7 @@ String _injectedClientSnippet(
       'window.\$dwdsDevHandlerPath = "$devHandlerPath";\n'
       'window.\$dwdsEnableDevtoolsLaunch = $enableDevtoolsLaunch;\n'
       'window.\$dartEntrypointPath = "$entrypointPath";\n'
+      'window.\$emitDebugEvents = "$emitDebugEvents";\n'
```
Comment (nshahan) [inline dwds/lib/src/handlers/injector.dart:208]: It looks like the naming convention for other variables attached to the window tries to identify where the dependency on them exists. `"$dart..."` for values used by the dart SDK. I think `"$dwds..."` are values used by the injected dwds client. I think this one should start with `"$dwds"

File: dwds/web/client.dart [hunk 2]
```diff
@@ -217,6 +219,9 @@ external bool get dwdsEnableDevtoolsLaunch;
 @JS('window.top.document.dispatchEvent')
 external void dispatchEvent(CustomEvent event);
 
+@JS(r'$emitDebugEvents')
+external bool get emitDebugEvents;
+
 @JS(r'$emitDebugEvent')
 external set emitDebugEvent(void Function(String, String) func);
```
Comment (nshahan) [inline dwds/web/client.dart:226]: nit: These two names are so close and since once is a getter and one is a setter it feels like they are a pair.

**Review by nshahan — APPROVED** [review]
I left a few comments.

Overall I think we just need to be careful where we disable this so it's only for targets in settings that are not depending on any devtools. AFAIK this will break the Angular and Flutter dev tools.

File: dwds/lib/src/handlers/injector.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/handlers/injector.dart:208]: Makes sense, will add!

Comment (annagrin) [reply dwds/lib/src/handlers/injector.dart]: done!

File: dwds/web/client.dart (hunk 2 above)
Comment (annagrin) [reply dwds/web/client.dart]: done!

**Comment by annagrin** [comment]
> AFAIK this will break the Angular and Flutter dev tools.

@nshahan do you know where are they used? I'd appreciate any pointers!

**Pull request merged**